source/simulator/simulator_access.cc

```diff
@@ -55,6 +55,13 @@ namespace aspect
     return *simulator;
   }
 
+  template <int dim>
+  SimulatorSignals<dim> &
+  SimulatorAccess<dim>::get_signals() const
+  {
+    // TODO: do not const_cast
+    return const_cast<Simulator<dim>*>(simulator)->signals;
```

Comment:
you could make it look less threatening if you didn't const-cast the simulator, but the signals variable. makes no actual difference, but looks less intimidating.

more in the general discussion.